### PR TITLE
[Enhancement] Add big query error in audit log

### PR DESCRIPTION
**Pull request by ZiheLiu**
## Why I'm doing:

When exceeding the thredsholds of `big_query_cpu_second_limit` or `big_query_scan_rows_limit`, the query will be failed. 
However, we don't know a query is killed by which big query thredsholds or other reasons.

## What I'm doing:

Add `BIG_QUERY_CPU_SECOND_LIMIT_EXCEEDED` and `BIG_QUERY_SCAN_ROWS_LIMIT_EXCEEDED` to `TStatusCode`.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

## Documentation PRs only:

If you are submitting a PR that adds or changes English documentation and have not
included Chinese documentation, then you can check the box to request GPT to translate the
English doc to Chinese. Please ensure to uncheck the **Do not translate** box if translation is needed.
The workflow will generate a new PR with the Chinese translation after this PR is merged.

- [ ] Yes, translate English markdown files with GPT
- [x] Do not translate

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

File: be/src/common/status.cpp [hunk 1]
```diff
@@ -261,6 +261,10 @@ std::string Status::code_as_string() const {
         return "Capaticy limit exceeded";
     case TStatusCode::SHUTDOWN:
         return "Shut down in progress";
+    case TStatusCode::BIG_QUERY_CPU_SECOND_LIMIT_EXCEEDED:
```
Comment (dirtysalt) [inline be/src/common/status.cpp:264]: do you have to handle big query memory limit?

**Comment by dirtysalt** [comment]
LGTM

File: be/src/common/status.cpp (hunk 1 above)
Comment (ZiheLiu) [reply be/src/common/status.cpp:264]: `big_query_mem_limit` is also handled by `TStatusCode:MEM_LIMIT_EXCEEDED`.

**Review by dirtysalt — APPROVED** [review]

**Comment by dirtysalt** [comment]
@alvin-celerdata do you think it's okay to merge? it's hard to cover resource group code.

File: gensrc/thrift/StatusCode.thrift [hunk 2]
```diff
@@ -112,5 +112,8 @@ enum TStatusCode {
     CAPACITY_LIMIT_EXCEED = 58,
 
     SHUTDOWN = 59, // the service is shutting down
+
+    BIG_QUERY_CPU_SECOND_LIMIT_EXCEEDED = 60,
+    BIG_QUERY_SCAN_ROWS_LIMIT_EXCEEDED = 61,
```
Comment (alvin-celerdata) [inline gensrc/thrift/StatusCode.thrift:117]: I'm a little confused about these items name. 
So do we have a concept of "BIG_QUERY"? Which kind of queries can be treated as BIG_QUERY?
And what do these two items mean?

Comment (dirtysalt) [reply gensrc/thrift/StatusCode.thrift:117]: [Resource group | StarRocks](https://docs.starrocks.io/docs/administration/management/resource_management/resource_group/)
https://docs.starrocks.io/docs/administration/management/resource_management/resource_group/

In resource group,  we have definition of "big query" by setting some limits: 

> On the basis of the above resource consumption restrictions, you can further restrict the resource consumption for big queries with the following parameters:
> 
> - big_query_cpu_second_limit: This parameter specifies the CPU upper time limit for a big query on a single BE. Concurrent queries add up the time. The unit is second. This parameter takes effect only when it is set greater than 0. Default: 0.
> - big_query_scan_rows_limit: This parameter specifies the scan row count upper limit for a big query on a single BE. This parameter takes effect only when it is set greater than 0. Default: 0.
> - big_query_mem_limit: This parameter specifies the memory usage upper limit for a big query on a single BE. The unit is byte. This parameter takes effect only when it is set greater than 0. Default: 0.
>

Comment (alvin-celerdata) [reply gensrc/thrift/StatusCode.thrift:117]: thanks for letting me know it.
Then I have no more questions.

**Review by alvin-phoenix-ai — APPROVED** [review]

**Pull request merged**